thorlcr/graph/thgraph.hpp

```diff
@@ -1134,6 +1136,8 @@ class graph_decl CActivityBase : implements CInterfaceOf<IThorRowInterfaces>, im
     inline void setInitialized(bool tf) { initialized = tf; }
     inline bool queryTimeActivities() const { return timeActivities; }
     inline roxiemem::RoxieHeapFlags queryHeapFlags() const { return defaultRoxieMemHeapFlags; }
+    inline bool traceStartStop() const { return false; }
+    inline bool traceActivity() const { return doTrace(traceThorActivities); }  // Could support overriding per activity via hint or other means, in the future
```

Comment:
there is an existing mechanism/function for per activity logging : container.doLogging()
However, at the moment it is:
a) always on for all graphs except child graphs
b) configurable per graph, not per activity (but would be easy to change so individual acts could be activated instead) :
```
void CGraphBase::init()
{
    bool log = queryJob().queryForceLogging(queryGraphId(), (NULL == queryOwner()) || isGlobal());
    setLogging(log);
}
```

This mechanism will still be active when traceThorActivities is enabled.